File: workspace/java/algorithms/datastructure/src/main/java/host/fairy/recursion/RSum.java
/*****************************************************
 * @software: IntelliJ IDEA
 * @author: Lionel Johnson
 * @contact: https://fairy.host
 * @organization: https://github.com/FairylandFuture
 * @datetime: 2025-09-10 15:00:23 UTC+08:00
 ****************************************************/
package host.fairy.recursion;

/**
 * @author Lionel Johnson
 * @version 1.0
 * @deprecated RSum: 递归求和
 */
public class RSum {
    public static Long sum(Long n) {
        if (n == 1) {
            return 1L;
        }

        return n + sum(n - 1);
    }

    public static void main(String[] args) {
        System.out.println(sum(1000000L));
    }
}

File: workspace/python/playground/playground-data-structure-algorithm/playground-01-data-structure/recursion/r_sum.py
# coding: UTF-8
"""
@software: PyCharm
@author: Lionel Johnson
@contact: https://fairy.host
@organization: https://github.com/FairylandFuture
@datetime: 2025-09-10 15:59:36 UTC+08:00
"""

import typing as t


def _sum(n: int, acc: int) -> int:
    if n == 1:
        return 1 + acc

    return _sum(n - 1, n + acc)


def main():
    print(_sum(10000000, 0))


if __name__ == "__main__":
    main()
